pkg/pxc/app/binlogcollector/binlog-collector.go

```diff
@@ -49,7 +50,7 @@ func GetService(cr *api.PerconaXtraDBCluster) *corev1.Service {
 	}
 }
 
-func GetDeployment(cr *api.PerconaXtraDBCluster, initImage string) (appsv1.Deployment, error) {
+func GetDeployment(cr *api.PerconaXtraDBCluster, initImage string, existingSelector *metav1.LabelSelector) (appsv1.Deployment, error) {
```

Comment:
Why are we passing the whole selector and not just the `MatchLabels` ? Do we also need to use the `MatchExpressions` for some reason? If we don't need it, we can simplify the logic and just pass what is needed i.e. the `MatchLabels map[string]string` as `existingMatchLabels`, WDYT?